client/post-editor/media-modal/index.jsx

```diff
@@ -109,24 +121,19 @@ export class EditorMediaModal extends Component {
 		};
 	}
 
-	isDisabled() {
-		return some( this.props.mediaLibrarySelectedItems, function( item ) {
-			const mimePrefix = MediaUtils.getMimePrefix( item );
-			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.props.view );
-		}.bind( this ) );
-	}
-
 	confirmSelection = () => {
 		const { view, mediaLibrarySelectedItems } = this.props;
 
-		let value;
-		if ( mediaLibrarySelectedItems.length ) {
-			value = {
+		if ( areMediaActionsDisabled( view, mediaLibrarySelectedItems ) ) {
```

Comment:
The `disabled` prop was apparently not enough to prevent the Insert action: the button would appear disabled but, on occasion, clicking it would still call `confirmSelection`. As a response, I first tried something like:

```diff
 			buttons.push( {
 				action: 'confirm',
 				label: this.props.labels.confirm || this.props.translate( 'Insert' ),
 				isPrimary: true,
 				disabled: isDisabled || 0 === selectedItems.length,
-				onClick: this.confirmSelection
+				onClick: isDisabled ? undefined : this.confirmSelection
 			} );
```

but in practice the behavior was quite clunky (can't remember the details).